### PR TITLE
makei c -f dir  is now supported

### PR DESCRIPTION
**Pull request by edmundreinhardt**
implements functionality requested by  issue #158

### CODE REVIEW

**Review by TongkunZhang — APPROVED** [review]

File: scripts/makei [hunk 1]
```diff
@@ -207,7 +207,15 @@ def handle_compile(args):
...
-    targets = get_compile_targets_from_filenames(filenames)
+    targets = []
+    source_names = []
+    for name in filenames:
+        if os.path.isdir(name):
+            targets.append(make_dir_target(name))
+        else:
+            source_names.append(name)
+    targets = targets.append(get_compile_targets_from_filenames(source_names))
+    print(targets)
```
Comment (jonnyz32) [inline scripts/makei:218]: Is this print here to display data to the user?

**Comment by jonnyz32** [comment]
@edmundreinhardt Would it be possible to also include in your pr a screenshot of a clean compilation of bob-recursive-example? And if possible, an addition to bob-recursive-example that will test this new functionality?

File: scripts/makei [hunk 2]
```diff
@@ -207,7 +207,15 @@ def handle_compile(args):
...
         filenames = map(os.path.basename, args.files.split(':'))
-    targets = get_compile_targets_from_filenames(filenames)
+    targets = []
+    source_names = []
+    for name in filenames:
+        if os.path.isdir(name):
+            targets.append(make_dir_target(name))
+        else:
+            source_names.append(name)
+    targets = targets.append(get_compile_targets_from_filenames(source_names))
```
[the remark this answers is not in the document]
Comment (edmundreinhardt) [reply scripts/makei]: good catch!  Fixed it in subsequent commit

**Comment by edmundreinhardt** [comment]
image
bob-recursive-example compiles clean

File: makei/build.py [hunk 3]
```diff
@@ -108,6 +108,8 @@ def map_ibmi_json_var(path):
         # if include path is not empty or *NONE then wrap in single quotes
         if len(includePath) > 0 and [v.upper() for v in includePath] != ["*NONE"]:
             incdir='\''+'\' \''.join(includePath)+'\''
+        elif len(includePath) == 1:
+            incdir=includePath[0].upper()
```
Comment (jonnyz32) [inline makei/build.py:112]: Do we need to capitalize the incdir only in the elif case? Also in the if case each individual path will have a set of quotes around it as well as a set of quotes to enclose the entire incdir, but in the elif case there will only be the enclosing quotes. Is this expected?

File: scripts/makei [hunk 4]
```diff
@@ -207,7 +207,17 @@ def handle_compile(args):
...
-    targets = get_compile_targets_from_filenames(filenames)
+    targets = []
+    source_names = []
+    for name in filenames:
+        if os.path.isdir(name):
+            targets.append(make_dir_target(name))
+        else:
+            source_names.append(name)
+    #print("source:"+' '.join(source_names))
+    #print("compile targets:"+' '.join(get_compile_targets_from_filenames(source_names)))
```
Comment (jonnyz32) [inline scripts/makei:218]: Can we remove these comments?

File: makei/build.py (hunk 3 above)
Comment (edmundreinhardt) [reply makei/build.py:112]: the other case does NOT have *NONE so we want the original case

File: scripts/makei (hunk 4 above)
Comment (edmundreinhardt) [reply scripts/makei:218]: These were here for debugging purposes, so I would prefer to leave them.

**Review by jonnyz32 — APPROVED** [review]

**Pull request merged**